judge/admin/problem.py

```diff
@@ -190,23 +187,20 @@ def _rescore(self, request, problem_id, publicy_changed=False):
         from judge.tasks import rescore_problem
         transaction.on_commit(rescore_problem.s(problem_id, publicy_changed).delay)
 
-    def update_publish_date(self, request, queryset):
-        count = queryset.update(date=timezone.now())
-        self.message_user(request, ngettext("%d problem's publish date successfully updated.",
-                                            "%d problems' publish date successfully updated.",
-                                            count) % count)
-
-    update_publish_date.short_description = _('Set publish date to now')
-
-    def make_public(self, request, queryset):
-        count = queryset.update(is_public=True)
+    def make_public_and_update_publish_date(self, request, queryset):
+        count = queryset.update(is_public=True, date=timezone.now())
         for problem_id in queryset.values_list('id', flat=True):
             self._rescore(request, problem_id, True)
+
         self.message_user(request, ngettext('%d problem successfully marked as public.',
                                             '%d problems successfully marked as public.',
                                             count) % count)
 
-    make_public.short_description = _('Mark problems as public')
+        self.message_user(request, ngettext("%d problem's publish date successfully updated.",
```

Comment:
could you remove this message? We only need the first one.

Reply:
Fixed.